src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -830,6 +849,47 @@ public void damage(double amount, Entity source, DamageCause cause) {
         setLastDamager(source);
     }
 
+    /**
+     * Checks if the incoming source of damage is a "Player Hit".
+     * @param source The incoming source of damage
+     */
+    private boolean isPlayerHit(Entity source) {
+        // If directly damaged by a player
+        if (source instanceof GlowPlayer) {
+            return true;
+        // If damaged by a TNT ignited by a player
+        } else if (source instanceof GlowTntPrimed) {
```

Comment:
The `else`s are redundant and can be omitted (return statement inside of the if-blocks).